--- algapi.h	2014-04-18 14:09:29.624398182 +0800
+++ ../algapi.h	2014-04-18 14:35:02.348396614 +0800
@@ -125,6 +125,7 @@
 extern const struct crypto_type crypto_ablkcipher_type;
 extern const struct crypto_type crypto_aead_type;
 extern const struct crypto_type crypto_blkcipher_type;
+extern const struct crypto_type crypto_pkc_type;
 
 void crypto_mod_put(struct crypto_alg *alg);
 
@@ -386,5 +387,17 @@
 	return (type ^ CRYPTO_ALG_ASYNC) & mask & CRYPTO_ALG_ASYNC;
 }
 
+static inline void *crypto_pkc_ctx(struct crypto_pkc *tfm)
+{
+	return crypto_tfm_ctx(&tfm->base);
+}
+
+/* RSA Request Completion handler */
+static inline void pkc_request_complete(struct pkc_request *req,
+					int err)
+{
+	req->base.complete(&req->base, err);
+}
+
 #endif	/* _CRYPTO_ALGAPI_H */
 
